source/Octopus.Tentacle/Commands/RegisterMachineCommandBase.cs

```diff
@@ -75,6 +76,7 @@ public abstract class RegisterMachineCommandBase<TRegistrationOperationType> : A
             Options.Add("server-comms-port=", "When using active communication, the comms port on the Octopus Server; the default is " + DefaultServerCommsPort + ". If specified, this will take precedence over any port number in server-comms-address.", s => serverCommsPort = int.Parse(s));
             Options.Add("server-comms-address=", "When using active communication, the comms address on the Octopus Server; the address of the Octopus Server will be used if omitted.", s => serverCommsAddress = s);
             Options.Add("server-web-socket=", "When using active communication over websockets, the address of the Octopus Server, eg 'wss://example.com/OctopusComms'. Refer to http://g.octopushq.com/WebSocketComms", s => serverWebSocketAddress = s);
+            Options.Add("server-subscription-id=", "When using active communication, the subscription id is what Octopus Server uses to identify this Tentacle this must be a valid URI e.g. poll://foobar/. If not specified, a new subscription id will be generated.", s => serverSubscriptionId = s);
```

Comment:
Is there a way to validate that this is `poll://ABC/`? SO people don't put junk here? Also, I think they are constrained to 20 chars?

Reply:
> Is there a way to validate that this is poll://ABC/

Currently it does enforce that it's a valid URI since we pass the value to the Uri constructor but we can take it a step further and validate that the scheme equals `poll`

> Also, I think they are constrained to 20 chars?

I thought so too but even though server mentions 20 characters it doesn't actually restrict it. It only fails when the length is longer than the maximum host name length
![Screenshot 2024-05-09 at 2 09 42 pm](https://github.com/OctopusDeploy/OctopusTentacle/assets/3272176/8c1f6809-282b-4c17-af2e-6d7b940d2367)